server/drivers/yard2LCD.c

```diff
@@ -0,0 +1,647 @@
+	debug(RPT_DEBUG, "%s: Event 08 - Enter yard_init",drvthis->name);
+	struct sockaddr_un srvAddr;
+	int tmp, srvAddrLen;
+	PrivateData *p;
+	char sockpath[200] = DEFAULT_SOCK_PATH; 
+	unsigned char byteCnt;
+	char Recbuffer[10];
+	
+
+	// Allocate and store private data
+	p = (PrivateData *)malloc(sizeof(PrivateData));
+	if (p == NULL)
+		return -1;
+	if (drvthis->store_private_ptr(drvthis, p) < 0)
+		return -1;
+
+	// Initialize cellsize, character spacing and graphical offset
+	p->fd = -1;
+	p->cellwidth = DEFAULT_CELLWIDTH;
+	p->cellheight = DEFAULT_CELLHEIGHT;
+	p->bigcellwidth = DEFAULT_CELLWIDTH_BIG;
+	p->bigcellheight = DEFAULT_CELLHEIGHT_BIG;
+	p->hspace = DEFAULT_HSPACE;
+	p->vspace = DEFAULT_VSPACE;
+	p->ccmode = standard;
+	p->dispmode = Dmode_standard;
+	p->LCDtype = 0; //only use 0=HD44780; not supported now 1=KS0066;2=T6963c;3=KS0108
+	p->width  = DEFAULT_WIDTH;
+	p->height = DEFAULT_HEIGHT;
+	p->gwidth = DEFAULT_GWIDTH;
+	p->gheight = DEFAULT_GHEIGHT;
+
+	// Establish connection to YARD server
+	bzero( (char *)&srvAddr, sizeof(srvAddr));
+	srvAddr.sun_family = AF_UNIX;
+	strcpy(srvAddr.sun_path, sockpath);
+	srvAddrLen = strlen(srvAddr.sun_path) + sizeof(srvAddr.sun_family);
+	
+	p->fd = socket(AF_UNIX, SOCK_STREAM, 0);
+	if (p->fd < 0) {
+		report(RPT_ERR, "%s: Can't create socket !", drvthis->name);
+		return -1;
+	}
+	if (connect(p->fd, (struct sockaddr *)&srvAddr, srvAddrLen) < 0) {
+		report(RPT_ERR, "%s: Can't connect to yardsrv !", drvthis->name);
+		return -1;
+	}
+	
+	//Get the config from yard2srvd. Config is not in LCDd.conf !
+	sprintf(Recbuffer,"LCDPROC\0");
+	byteCnt = write(p->fd,Recbuffer, strlen(Recbuffer)); 
+	if (byteCnt < 0) 
+	{
+		report(RPT_ERR, "%s: Can't send config request to YARD2 LCDserver !", drvthis->name);
+		return -1;
+	}
+
+	// Wait for YARD server response
+	byteCnt = read(p->fd, Recbuffer, sizeof(Recbuffer) );
+	if (byteCnt < 0) 
+	{
+		report(RPT_ERR, "%s: Cannot receive config from YARD server !", drvthis->name);
+		return -1;
+	}
+	else if (byteCnt == 1) 
+	{
+		report(RPT_ERR, "%s: YARD communication timeout !", drvthis->name);
+		return -1;
+	}
+
+	//Receiveconfig format: 0='C';1=sizeX;2=sizeY;3=LCDType(0=HD44780,1=KS0066,2=T6963,3=KS0108)
+	if(Recbuffer[0] == 'C') //C=Config
+	{
+		p->width   = Recbuffer[1];
+		p->height  = Recbuffer[2];
+		p->LCDtype = Recbuffer[3];
+	}
+	else
+	{
+		report(RPT_ERR, "%s: YARD Config Receive error !", drvthis->name);
+		return -1;
+	}
+
+	if(p->LCDtype > 1) //only char LCDs are currently supported
+	{
+		report(RPT_ERR, "%s: YARD LCD type %d not supported by this version or the driver !", drvthis->name,p->LCDtype);
+		return -1;
+	}
+	
+	// Allocate framebuf & Setup frame buffer x2 to be sure that the buffer is big enough
+	p->framebuf = (unsigned char *) malloc((p->width * p->height)*2);
+	if (p->framebuf == NULL) 
+	{
+		report(RPT_ERR, "%s: Can't create framebuffer !", drvthis->name);
+		return -1;
+	}
+	memset(p->framebuf, ' ', p->width * p->height);
+
```

Comment:
Why do you need a double size frame buffer but then only initialize the first half?